pimd/pim_cmd_common.c

```diff
@@ -2851,12 +2853,21 @@ static int pim_print_json_pnc_cache_walkcb(struct hash_bucket *backet,
 		json_ifp = json_object_new_object();
 		json_object_string_add(json_ifp, "interface",
 				       ifp ? ifp->name : "NULL");
+
+		if (ifp)
+			pim_ifp = ifp->info;
+
+		if (pim_ifp && pim_ifp->pim_enable)
+			pim_enable = true;
+
 #if PIM_IPV == 4
 		json_object_string_addf(json_ifp, "nexthop", "%pI4",
 					&nh_node->gate.ipv4);
+		json_object_boolean_add(json_ifp, "pimEnabled", pim_enable);
 #else
 		json_object_string_addf(json_ifp, "nexthop", "%pI6",
 					&nh_node->gate.ipv6);
+		json_object_boolean_add(json_ifp, "pim6Enabled", pim_enable);
```

Comment:
I think it is better to just add pimEnabled field for both pimv4 and pimv6 since the protocol is just PIM and not PIMv6 and since the CLIs for pimv4 and pimv6 are already different for both pimv4 and pimv6, so keeping pimEnabled won't create any confusion.